tests/runtime/conftest.py

```diff
@@ -138,17 +150,59 @@ def runtime_grpc_test_server(open_port, *args, **kwargs):
 
 @pytest.fixture(scope="session")
 def runtime_grpc_server(
-    session_scoped_open_port, sample_inference_service, sample_train_service
+    session_scoped_open_port,
 ) -> RuntimeGRPCServer:
     with runtime_grpc_test_server(
         session_scoped_open_port,
-        inference_service=sample_inference_service,
-        training_service=sample_train_service,
     ) as server:
         _check_server_readiness(server)
         yield server
 
 
+@contextmanager
+def runtime_http_test_server(open_port, *args, **kwargs):
+    """Helper to wrap creation of RuntimeHTTPServer in temporary configurations"""
+    with tempfile.TemporaryDirectory() as workdir:
+        temp_log_dir = os.path.join(workdir, "metering_logs")
+        temp_save_dir = os.path.join(workdir, "training_output")
+        os.makedirs(temp_log_dir)
+        os.makedirs(temp_save_dir)
+        with temp_config(
+            {
+                "runtime": {
+                    "metering": {"log_dir": temp_log_dir},
+                    "training": {"output_dir": temp_save_dir},
+                    "http": {"port": open_port},
+                }
+            },
+            "merge",
+        ):
+            config_overrides = {}
+            if "tls_config_override" in kwargs:
+                config_overrides = kwargs["tls_config_override"]
+                kwargs["tls_config_override"] = config_overrides["runtime"]["tls"]
+            with http_server.RuntimeHTTPServer(*args, **kwargs) as server:
+                _check_http_server_readiness(server, config_overrides)
+                # Give tests access to the workdir
+                server.workdir = workdir
+                yield server
+
+
+# This is honestly not required for fastAPI testing
```

Comment:
If it's not required and not used, I'd be in favor of removing it to avoid unused code